client/types.ts

```diff
@@ -31,6 +31,13 @@ export type TimeoutMS = NonUndefined< Parameters< typeof setTimeout >[ 1 ] >;
 export type TimestampMS = ReturnType< typeof Date.now >;
 export type TimerHandle = ReturnType< typeof setTimeout >;
 export type IntervalHandle = ReturnType< typeof setInterval >;
+export type JSONSerializable =
+	| null
+	| boolean
+	| number
+	| string
+	| JSONSerializable[]
+	| { [ prop: string ]: JSONSerializable };
```

Comment:
cc: @sgomes I think at some point you and I spoke of adding this type. could be useful in other places, like in reducer persistence